_posts/2023-11-09-in-progress-wildfly-feature-on-openshift.adoc

```diff
@@ -0,0 +1,298 @@
+```
+docker build -t quay.io/<your-quay-username>/wildfly:wildfly-app .
+docker push quay.io/<your-quay-username>/wildfly:wildfly-app
+```
+Replace `<your-quay-username>` with your username for Quay. The `-t` command indicates that we are specifying a tag, which in this case is `wildfly-app`. The tag is used to identify different images inside the same repository. Using this image, we can ensure that we are running your local version of WildFly, as opposed to the version of WildFly that is present on OpenShift. 
+
+== Build the Application 
+
+Now that we have our WildFly container image, we can use it to deploy our application on it. Navigate to the `elytron-examples/elytron-oidc-client-scope` directory and create a deployment file for your application using this command: 
+```
+mvn clean package
+```
+This will create a `simple-webapp-oidc.war` file inside the `target` directory. 
+
+Next, create a new `Dockerfile` inside the `elytron-oidc-client-scope` project with the following commands inside it: 
+```
+FROM quay.io/<your-quay-username>/wildfly:wildfly-app
+ADD target/simple-webapp-oidc.war /opt/server/standalone/deployments/
+```
+Here, we are making use of the container image of WildFly we made earlier and copying over the deployment file to the container image so we can deploy it on OpenShift by running the server. Next, build and push a new image of your application using the following commands: 
+```
+docker build -t quay.io/<your-quay-username>/wildfly:latest .
+docker push quay.io/<your-quay-username>/wildfly:latest
+```
+=== Configure Image Pull Secret
+
+In order to pull the image from Quay, you will need to configure a pull secret and connect it to your helm release. If you have already configured a pull secret for Quay, then you can skip this step.
+
+Navigate to your Quay console, click on `Account Settings`, `Generate Encrypted Password` and log in. Under the `Kubernetes Secret` tab you will be able to download the yaml file for your Kubernetes secret. Upload it to your OpenShift Secrets by going to the `Secret` tab on Developer Sandbox, clicking on `Create`, `From Yaml` and replacing it with the contents of your Kubernetes Secret. 
+
+Alternatively you can link your pull secret so OpenShift can automatically use your pull secret to obtain the image when needed. You can learn more about handling secrets on OpenShift on https://docs.openshift.com/container-platform/4.10/openshift_images/managing_images/using-image-pull-secrets.html[their documentation]. 
+
+== Add Helm Configuration
+
+Lastly, we will be creating a helm release of our local WildFly image and deploying our application on it using a helm chart. Navigate to `charts` directory and open the `helm.yaml` file. 
+```
+cd /PATH/TO/ELYTRON/EXAMPLES/elytron-oidc-client-scope/charts
+```
+The contents of the file will be as follows: 
+```
+image: 
+  name: quay.io/<your-quay-username>/wildfly
+build:
+  enabled: false
+deploy:
+  env:
+    - name: OIDC_PROVIDER_URL
```

Comment:
I am working with this file: https://github.com/PrarthonaPaul/elytron-examples/blob/elytron-oidc-client-scope/elytron-oidc-client-scope/charts/helm.yaml and the contents are a bit different than copied here?

Tthe value of OIDC_PROVIDER_URL has a different hash each time probably, so I would parametrize the full value not just \<your-project-name\>

Reply:
should be good now.